src/Parser/DemonHunter/Havoc/Modules/ResourceTracker/FuryTracker.js

```diff
@@ -0,0 +1,67 @@
+import Combatants from 'Parser/Core/Modules/Combatants';
+import RESOURCE_TYPES from 'common/RESOURCE_TYPES';
+import ResourceTracker from 'Parser/Core/Modules/ResourceTracker/ResourceTracker';
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+
+class FuryTracker extends ResourceTracker {
+	static dependencies = {
+		combatants: Combatants,
+		spellUsable: SpellUsable,
+	};
+
+	totalCooldownReduction = 0;
+	totalCooldownReductionWasted = 0;
+
+	on_initialized() {
+		this.resourceType = RESOURCE_TYPES.FURY.id;
+		this.resourceName = 'Fury';
+	}
+
+	getReducedCost(event) {
+		if(!this.getResource(event).cost) {
+			return 0;
+		}
+		let cost = this.getResource(event).cost;
+		const spellId = event.ability.guid;
+		if ((spellId === SPELLS.BLADE_DANCE.id || spellId === SPELLS.DEATH_SWEEP.id) 
+			 && (this.combatants.selected.hasTalent(SPELLS.FIRST_BLOOD_TALENT.id) || this.combatants.selected.hasRing(SPELLS.SOUL_OF_THE_SLAYER.id))) {
+			cost = cost - 20;
+		}
+		if(spellId === SPELLS.EYE_BEAM.id) {
+			const rank = this.combatants.selected.traitsBySpellId[SPELLS.WIDE_EYES.id];
+			cost = cost - 5 * rank;
+		}
+		if(spellId === SPELLS.CHAOS_NOVA.id && this.combatants.selected.hasTalent(SPELLS.UNLEASHED_POWER_TALENT.id)) {
+			cost = 0;
+		}
+		this.reduceCooldown(cost);
+		return cost;
+	}
+	
+
+	reduceCooldown(cost) {
+		if(!this.combatants.selected.hasShoulder(ITEMS.DELUSIONS_OF_GRANDEUR.id)){
+			return;
+		}
+		const COOLDOWN_REDUCTION_MS = 1000/30;
```

Comment:
This should probably a `const` outside the class (or even read from the legendary's file)

Reply:
I tried putting it in the DoG file and passing it into fury tracker but got a call stack error